UnityProject/Assets/Scripts/Core/SecureStuff/URLOpen/SafeURLOpen.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using UnityEngine;
+
+public static class SafeURL
+{
+
+	public static void Open(string URL)
+	{
+		URL = URL.ToLower();
+
+		if (URL.StartsWith($"https://"))
```

Comment:
don't parse this yourself, use URIs instead. https://learn.microsoft.com/en-us/dotnet/api/system.uri?view=netframework-4.8.1

Reply:
does later on in the function

Reply:
get and check if the protocol is valid using an URI, do not parse URLs into strings to do validation.